File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        res = []
        dc = {}
        for i in range(len(nums)):
            diff = target-nums[i]
            if diff in dc:
                res.append(dc[diff])
                res.append(i)
                return res
            dc[nums[i]]=i
        
        return res

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let res = []
    let dict = {}

    for (let i=0; i<nums.length; i++){
        diff = target-nums[i]
        if (diff in dict){
            res.push(dict[diff])
            res.push(i)
            break
        }
        dict[nums[i]]=i
    }
    return res
};